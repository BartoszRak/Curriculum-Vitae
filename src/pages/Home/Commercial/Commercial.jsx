import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { Typography, Container } from '@material-ui/core'
import { injectIntl, FormattedMessage } from 'react-intl'

import TileTriangle from '~assets/icons/TileTriangle'
import ProjectsList from '~components/ProjectsList'

import useStyle from './Commercial.style'
import technologiesMap from '~services/Technologies/TechnologiesMap.json'

export function Commercial({ classes: overridingClasses, intl }) {
  const classes = { ...useStyle(), ...overridingClasses }
  const { formatMessage } = intl
  const commercialProjectsArray = [
    {
      description: formatMessage({
        id: 'pages.home.commercial.projects.learningManagementSystem.description',
      }),
      tags: [
        technologiesMap.react,
        technologiesMap.redux,
        technologiesMap.rematch,
        technologiesMap.materialUi,
        technologiesMap.immutable,
      ],
      title: formatMessage({
        id: 'pages.home.commercial.projects.learningManagementSystem.title',
      }),
    },
    {
      description: formatMessage({
        id: 'pages.home.commercial.projects.quizzesManagementSystem.description',
      }),
      tags: [
        technologiesMap.react,
        technologiesMap.redux,
        technologiesMap.rematch,
        technologiesMap.materialUi,
        technologiesMap.firebase,
        technologiesMap.recharts,
        technologiesMap.jest,
      ],
      title: formatMessage({
        id: 'pages.home.commercial.projects.quizzesManagementSystem.title',
      }),
    },
    {
      description: formatMessage({
        id: 'pages.home.commercial.projects.quizzesManagementSystem.description',
      }),
      tags: [
        technologiesMap.react,
        technologiesMap.redux,
        technologiesMap.rematch,
        technologiesMap.materialUi,
        technologiesMap.firebase,
        technologiesMap.jest,
      ],
      title: formatMessage({
        id: 'pages.home.commercial.projects.quizzesManagementSystem.title',
      }),
    },
    {
      description: formatMessage({
        id: 'pages.home.commercial.projects.loadTestingPlatform.description',
      }),
      tags: [
        technologiesMap.react,
        technologiesMap.materialUi,
        technologiesMap.graphql,
        technologiesMap.testingLibrary,
      ],
      title: formatMessage({
        id: 'pages.home.commercial.projects.loadTestingPlatform.title',
      }),
    },
  ]

  return (
    <React.Fragment>
      <TileTriangle className={classNames(classes.triangle, classes.topTriangle)} />
      <section className={classes.section}>
        <Container classes={{ root: classes.root }} id="commercial">
          <Typography className={classes.title} color="secondary" variant="h1">
            <FormattedMessage id="pages.home.commercial.title" />
            <span aria-hidden className={classes.emoji}>
              {' '}
              ü¶∏‚Äç‚ôÇÔ∏è
            </span>
          </Typography>
          <Typography className={classes.subtitle} color="secondary" variant="h2">
            <FormattedMessage id="pages.home.commercial.subtitle" />
          </Typography>
          <ProjectsList data={commercialProjectsArray.reverse()} />
        </Container>
      </section>
      <TileTriangle className={classNames(classes.triangle, classes.bottomTriangle)} />
    </React.Fragment>
  )
}

Commercial.propTypes = {
  classes: PropTypes.object,
  intl: PropTypes.object,
}

export default injectIntl(Commercial)
