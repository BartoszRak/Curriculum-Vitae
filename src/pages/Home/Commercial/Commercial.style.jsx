import { makeStyles } from '@material-ui/core'

export default makeStyles(({ breakpoints, palette }) => ({
  bottomTriangle: {
    marginTop: '-1px',
    transform: 'rotate(180deg)',
  },
  emoji: {
    fontSize: '2rem',
    [breakpoints.down('md')]: {
      fontSize: '1.5rem',
    },
  },
  root: {
    width: '100%',
  },
  section: {},
  subtitle: {
    color: palette.primary.main,
    [breakpoints.down('md')]: {
      fontSize: '1.8rem',
    },
  },
  title: {
    color: palette.primary.main,
    [breakpoints.down('md')]: {
      fontSize: '2rem',
    },
  },
  triangle: {
    color: palette.secondary.main,
  },
}))
