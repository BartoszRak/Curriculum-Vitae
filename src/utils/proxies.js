/* eslint-disable */
export const createPropertyProxy = (obj = {}) => new Proxy(obj, {
  get: (target, name) => target[name] || `${name}Mock`,
})

export const PropertyProxy = createPropertyProxy()
