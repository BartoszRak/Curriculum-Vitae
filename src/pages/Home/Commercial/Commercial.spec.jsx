import React from 'react'
import { cleanup } from '@testing-library/react'
import { shallow } from 'enzyme'

import { Commercial } from './Commercial'
import { PropertyProxy } from '~utils/proxies'

jest.unmock('@material-ui/core')

jest.mock('~components/ProjectsList', () => 'ProjectsListMock')

let props

describe('component: Commercial', () => {
  afterEach(cleanup)

  beforeEach(() => {
    props = {
      classes: PropertyProxy,
      intl: {
        formatMessage: jest.fn(value => value),
      },
    }
  })
  describe('rendering', () => {
    test('render without crash', () => {
      const wrapper = shallow(<Commercial {...props} />)

      expect(wrapper).toBeTruthy()
    })

    test('match snapshot', () => {
      const wrapper = shallow(<Commercial {...props} />)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
