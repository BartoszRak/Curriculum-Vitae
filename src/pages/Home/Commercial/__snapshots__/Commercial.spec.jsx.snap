// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Commercial rendering match snapshot 1`] = `
<Fragment>
  <TileTriangle
    className="makeStyles-triangle-7"
  />
  <section
    className="makeStyles-section-4"
  >
    <WithStyles(ForwardRef(Container))
      classes={
        Object {
          "root": "makeStyles-root-3",
        }
      }
      id="commercial"
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-6"
        color="secondary"
        variant="h1"
      >
        <FormattedMessage
          id="pages.home.commercial.title"
          values={Object {}}
        />
        <span
          aria-hidden={true}
          className="makeStyles-emoji-2"
        >
           
          ü¶∏‚Äç‚ôÇÔ∏è
        </span>
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-subtitle-5"
        color="secondary"
        variant="h2"
      >
        <FormattedMessage
          id="pages.home.commercial.subtitle"
          values={Object {}}
        />
      </WithStyles(ForwardRef(Typography))>
      <ProjectsListMock
        data={
          Array [
            Object {
              "description": Object {
                "id": "pages.home.commercial.projects.loadTestingPlatform.description",
              },
              "tags": Array [
                Object {
                  "name": "React",
                  "url": "https://reactjs.org/",
                },
                Object {
                  "name": "Material UI",
                  "url": "https://material-ui.com/",
                },
                Object {
                  "name": "GraphQL",
                  "url": "https://graphql.org/",
                },
                Object {
                  "name": "Testing library",
                  "url": "https://testing-library.com/",
                },
              ],
              "title": Object {
                "id": "pages.home.commercial.projects.loadTestingPlatform.title",
              },
            },
            Object {
              "description": Object {
                "id": "pages.home.commercial.projects.quizzesManagementSystem.description",
              },
              "tags": Array [
                Object {
                  "name": "React",
                  "url": "https://reactjs.org/",
                },
                Object {
                  "name": "Redux",
                  "url": "https://redux.js.org/",
                },
                Object {
                  "name": "Rematch",
                  "url": "https://github.com/rematch/rematch",
                },
                Object {
                  "name": "Material UI",
                  "url": "https://material-ui.com/",
                },
                Object {
                  "name": "Firebase",
                  "url": "https://firebase.google.com/",
                },
                Object {
                  "name": "Jest",
                  "url": "https://jestjs.io/",
                },
              ],
              "title": Object {
                "id": "pages.home.commercial.projects.quizzesManagementSystem.title",
              },
            },
            Object {
              "description": Object {
                "id": "pages.home.commercial.projects.learningManagementSystem.description",
              },
              "tags": Array [
                Object {
                  "name": "React",
                  "url": "https://reactjs.org/",
                },
                Object {
                  "name": "Redux",
                  "url": "https://redux.js.org/",
                },
                Object {
                  "name": "Rematch",
                  "url": "https://github.com/rematch/rematch",
                },
                Object {
                  "name": "Material UI",
                  "url": "https://material-ui.com/",
                },
                Object {
                  "name": "Immutable",
                  "url": "https://github.com/immutable-js/immutable-js",
                },
              ],
              "title": Object {
                "id": "pages.home.commercial.projects.learningManagementSystem.title",
              },
            },
          ]
        }
      />
    </WithStyles(ForwardRef(Container))>
  </section>
  <TileTriangle
    className="makeStyles-triangle-7 makeStyles-bottomTriangle-1"
  />
</Fragment>
`;
